File: GrpProj1/BlackjackJS/Dealer.js
/*
	File: Dealer.js
	Author: Kevin Romero
	Purpose: Dealer class for Blackjack

*/

//Create dealer object
function Dealer(){

};

//Create object reference
Dealer.prototype = Object.create(new Player());

//constructor
Dealer.prototype.constructor=function(shoe){
	this.aShoe=shoe;
	this.isBust();
};

//print dealer hand
Dealer.prototype.prntDealerHand=function(){
	this.aHand.printHand();
};

//print dealer deck
Dealer.prototype.prntHouseDeck=function(){
	this.viewShoe();
};

//implement gameplay
Dealer.prototype.game=function(p){
	while(this.aHand.getHandTotal() <= 21){
		this.prntDealerHand();
		document.write("<br>");

		if(this.aHand.getHandTotal() > p.getTotScore() &&
			this.aHand.getHandTotal() <= 21){
			document.write("The House Wins! <br>");
			document.write("House Score was : "+
				this.aHand.getHandTotal()+"<br>");
			return;
		}
		if(this.aHand.getHandTotal() == p.getTotScore()){
			document.write("Tied! <br>");
			return;
		}
		this.hit();
	}
	if(this.aHand.getHandTotal() > 21){
		document.write("House Busted! You Won! <br>");
		return;
	}
};

File: GrpProj1/Blackjack code in C++/Blackjack-master/Dealer.cpp
/* 
 * File:   Dealer.h
 * Author: Kevin Romero
 * Purpose: Dealer Class for Blackjack
 * Created on July 13, 2017, 3:50 PM
 */

//System Libraries

//User Libraries
#include "Dealer.h"

//Implement default constructor
Dealer::Dealer(Shoe *s) : Player(s) {
    
}

//Print dealer hand
void Dealer::prntDealerHand(){
    aHand.printHand();
}

//Print dealer deck
void Dealer::prntHouseDeck(){
    Player::viewShoe();
}

//Implement gameplay
void Dealer::game(Player &p){
    while(aHand.getHandTotal() <= 21){
        this->prntDealerHand();
        //this->hit();
        cout<<"\n";
        
        if(aHand.getHandTotal() > p.getTotScore() && aHand.getHandTotal() <= 21){
            cout<<"The House Wins! \n";
            cout<<"House Score was : "<<aHand.getHandTotal()<<"\n";
            return;
        }
        if(aHand.getHandTotal() == p.getTotScore()){
            cout<<"Tied! \n";
            return;
        }
        this->hit();
    }
        /*if(aHand.getHandTotal() < p.getTotScore() && aHand.getHandTotal() <=21){
            while(aHand.getHandTotal() <= 16){
                this->hit();
            }
        }*/
  
    if(aHand.getHandTotal() > 21){
        cout<<"House Busted! You Won! \n";
        return;
    }
}
    